Fix cosmetic issues with the split payment and verification flow

rlucioni opened the pull request:
Use honor code ribbon if verification status is not missing, submitted, or approved, and more appropriately title the column on the payment confirmation page holding dollar amounts. Resolves [ECOM-878](https://openedx.atlassian.net/browse/ECOM-878) and [ECOM-881](https://openedx.atlassian.net/browse/ECOM-881), respectively.

@AlasdairSwan and @wedaly, could you review this when you're able?

wedaly commented on lms/templates/dashboard/_dashboard_course_listing.html:
```diff
@@ -29,7 +29,19 @@
+      % endif
+    % else:
+      <article class="course ${enrollment.mode}">
+    % endif
```
This makes sense to me, but I think you'll need to handle the "professional" case as well.

Could you please add test cases to `student/tests/test_verification_status`?  I think this logic is sufficiently complex (and easy enough to break) that we should be testing it.

rlucioni replied:
From offline discussions with @wedaly and @griffresch: we don't currently use the verification status to manipulate the banner message for professional education courses; this is not a concern for now, and as such we won't manipulate the banner color for these courses either.

AlasdairSwan replied:
Would it be cleaner to define a class variable that we then pass in to the article element so we only have to write <element class="course"> once?

rlucioni replied:
Sure, I think using a variable would make sense. That would leave us writing the <article> element twice, as before, instead of five times.

AlasdairSwan replied:
Can we not default the variable to an empty string so we only need to write the element once?

rlucioni replied:
![](http://managersmagazine.com/wp-content/uploads/2011/05/obama-yes-we-can.jpg)

rlucioni commented:
@AlasdairSwan @wedaly all set here. I've added test cases, double-checked on the status of the prof ed banner with @griffresch, and cleaned up the templating logic a bit.

AlasdairSwan commented:
:trumpet:

wedaly commented:
:+1:  thanks for cleaning this up!
